### PR TITLE
feat(api): add model for club roles

### PR DESCRIPTION
**Pull request by PranshuNayak**

### CODE REVIEW

**Comment by PranshuNayak** [comment]
@Samy-33 raised the PR

File: api/api/Roles/models.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    CLUB_ROLE_TYPES = [
+        ('Coordinator', 'Coordinator'),
+        ('Co-Coordinator', 'Co-Coordinator'),
+        ('Faculty Incharge', 'Faculty Incharge'),
+        ('Convener', 'Convener'),
+        ('Co-convener', 'Co-convener'),
+        ('Conselor', 'Conselor'),
+        ('Associate-Conselor', 'Associate-Conselor')
+    ]
```
Comment (Aksh-Bansal-dev) [inline api/api/Roles/models.py:16]: We should also add a role for core members as it is not necessary that only the coco or the coordinator will host an event.

File: api/api/Roles/models.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    name = models.CharField(max_length=30, choices=CLUB_ROLE_TYPES)
+    club = models.OneToOneField(Club, on_delete=models.CASCADE)
+    user = models.OneToOneField(User, on_delete=models.CASCADE)
```
Comment (Samy-33) [inline api/api/Roles/models.py:21]: Why are these `OneToOneFields`? A club can have multiple roles, and a user can have multiple roles as well.

**Review by Samy-33 — CHANGES_REQUESTED** [review]

File: api/api/Roles/apps.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+from django.apps import AppConfig
+
+
+class EventsConfig(AppConfig):
+    default_auto_field = 'django.db.models.BigAutoField'
+    name = 'api.Roles'
+    label = 'api_Roles'
```
Comment (Samy-33) [inline api/api/Roles/apps.py:7]: name and label must be lowercase.

**Review by Samy-33 — CHANGES_REQUESTED** [review]

File: api/api/Roles/models.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        ('Faculty Incharge', 'Faculty Incharge'),
+        ('Convener', 'Convener'),
+        ('Co-convener', 'Co-convener'),
+        ('Conselor', 'Conselor'),
+        ('Associate-Conselor', 'Associate-Conselor')
+    ]
+
+    name = models.CharField(max_length=30, choices=CLUB_ROLE_TYPES)
+    club = models.ManyToManyField(Club, on_delete=models.CASCADE)
+    user = models.ManyToManyField(User, on_delete=models.CASCADE)
```
Comment (Samy-33) [inline api/api/Roles/models.py:21]: These should be `ForeignKeyFields`. Why?

club as a `ManyToManyField` means, 1 object of club will have multiple objects of `Role` class, and 1 object of `Role` will have multiple clubs associated with it.

Do we want that? No.

One object of `Role` class means: one relationship between club, user and the role_name. Hence `ForeignKeyField` makes more sense.

**Review by Samy-33 — CHANGES_REQUESTED** [review]

**Review by Samy-33 — APPROVED** [review]
LGTM

**Pull request merged**